pgx-version-updater/src/main.rs

```diff
@@ -0,0 +1,407 @@
+
+fn main() {
+    let args = Args::parse();
+    let current_dir = env::current_dir().expect("Could not get current directory!");
+
+    // Contains a set of package names (e.g. "pgx", "pgx-pg-sys") that wil be used
+    // to search for updatable dependencies later on
+    let mut updatable_package_names_set: HashSet<String> = HashSet::new();
+
+    // This will eventually contain every file we want to process
+    let mut files_to_process_set: HashSet<String> = HashSet::new();
+
+    // Keep track of which files to exclude from a "package version" change.
+    // For example, some Cargo.toml files do not need this updated:
+    //   [package]
+    //   version = "0.1.0"
+    //   ...
+    // Any such file is explicitly added via a command line argument.
+    // Note that any files included here are still eligible to be processed for
+    // *dependency* version updates.
+    let mut exclude_version_files_set: HashSet<String> = HashSet::new();
+    for file in args.exclude_from_version_change {
+        exclude_version_files_set.insert(fullpath(file).to_str().unwrap().to_string());
+    }
+
+    // Recursively walk down all directories to extract out any existing Cargo.toml files
+    for entry in WalkDir::new(&current_dir)
+        .into_iter()
+        .filter_entry(|e| is_not_excluded_dir(e))
+        .filter_map(|v| v.ok())
+    {
+        if is_cargo_toml_file(&entry) {
+            let filepath: String = fullpath(entry.path().display().to_string())
+                .display()
+                .to_string();
+
+            let mut output = format!(
+                "{} Cargo.toml file at {}",
+                "Discovered".bold().green(),
+                &filepath.cyan()
+            );
+
+            // Extract the package name if possible
+            if !exclude_version_files_set.contains(&filepath) {
+                match extract_package_name(&filepath) {
+                    Some(package_name) => {
+                        updatable_package_names_set.insert(package_name);
+                    }
+                    None => {
+                        output.push_str(
+                            "\n           * Could not determine package name due to [package] not existing -- skipping version bump."
+                                .dimmed()
+                                .to_string()
+                                .as_str(),
+                        )
+                    }
+                }
+            }
+
+            files_to_process_set.insert(filepath.clone());
+            println!("{output}");
+        }
+    }
+
+    // Loop through all files that are included for dependency updates via CLI params
+    for file in args.include_for_dep_updates {
+        let filepath: String = fullpath(file).display().to_string();
+
+        let mut output = format!(
+            "{} Cargo.toml file at {} for processing",
+            " Including".bold().green(),
+            &filepath.cyan()
+        );
```

Comment:
This level of chattiness should definitely be `--verbose`-gated, as the user already knows what they provided. Uh. We hope.